zkevm-circuits/src/evm_circuit/execution/end_tx.rs

```diff
@@ -203,6 +250,23 @@ impl<F: Field> ExecutionGadget<F> for EndTxGadget<F> {
             coinbase_balance,
         )?;
 
+        let current_cumulative_gas_used: u64 = if tx.id == 1 {
+            0
+        } else {
+            let rw = &block.rws[(RwTableTag::TxReceipt, (tx.id - 1) * 3 - 1)];
```

Comment:
I think we should refactor this line. In case the `TxReceiptField` has more variants, this line would lead to bugs in assignment.

Something more like:
```rust
let rw = block.rws.tx_receipt(tx.id, TxReceiptField::CumulativeGasUsed);
```

What do you think?

Reply:
I am not really got what is the concerned issue :),  Here is to get previous tx 's `CumulativeGasUsed`,  since each tx receipt has three rows corresponding to three tags (PostStateOrStatus,  CumulativeGasUsed,  LogLength).  Glad to  know if you can make it more clear !

Reply:
I meant that this part: `(tx.id - 1) * 3 - 1` looks a bit messy and prone to errors in case the number of fields is edited in the future. Hence to refactor this would make sense.

Reply:
I see,  will try to use fields number here, like            ` let rw = &block.rws[(RwTableTag::TxReceipt, (tx.id - 1) * TxReceiptFieldTag::amount() - 1)];`.

Reply:
updated in https://github.com/appliedzkp/zkevm-circuits/pull/466/commits/db106a33e264ff347da007fa2028dfc9ee8d0063 thanks